ui/push-health/Test.jsx

```diff
@@ -26,6 +48,102 @@ class Test extends PureComponent {
     }
   }
 
+  addSelectedTest = (test) => {
+    this.setState((prevState) => ({
+      selectedTests: prevState.selectedTests.add(test),
+    }));
+  };
+
+  removeSelectedTest = (test) => {
+    const { selectedTests } = this.state;
+    selectedTests.delete(test);
+    this.setState({
+      selectedTests,
+    });
+  };
+
+  retriggerSelected = (times) => {
+    const { notify, currentRepo, jobs } = this.props;
+    const { selectedTests } = this.state;
+
+    // Reduce down to the unique jobs
+    const testJobs = Array.from(selectedTests).reduce(
+      (acc, test) => ({
+        ...acc,
+        ...jobs[test.jobName].reduce((fjAcc, job) => ({ [job.id]: job }), {}),
+      }),
+      {},
+    );
+    const uniqueJobs = Object.values(testJobs);
+
+    JobModel.retrigger(uniqueJobs, currentRepo, notify, times);
+  };
+
+  markAsInvestigated = () => {
+    const { selectedTests } = this.state;
+    const { notify, currentRepo, revision, updatePushHealth } = this.props;
+
+    let data;
+    let failureStatus;
+    if (selectedTests.size === 0) {
+      notify(`Select atleast one test`, 'warning');
+    } else {
+      selectedTests.forEach(async (test) => {
+        ({ data, failureStatus } = await create(
+          `${getProjectUrl(
+            investigatedTestsEndPoint,
+            currentRepo.name,
+          )}?revision=${revision}`,
+          {
+            test: test.testName,
+            jobName: test.jobName,
+            jobSymbol: test.jobSymbol,
+          },
+        ));
+        if (failureStatus) {
+          notify(data, 'warning');
+        } else {
+          selectedTests.delete(test);
+          this.setState({ selectedTests });
+          updatePushHealth();
+          notify(`Test ${data.test}  marked as investigated`, 'success');
+        }
+      });
+    }
+  };
+
+  markAsUninvestigated = () => {
+    const { selectedTests } = this.state;
+    const { notify, currentRepo, revision, updatePushHealth } = this.props;
+
+    if (selectedTests.size === 0) {
+      notify(`Select atleast one test`, 'warning');
+    } else {
+      selectedTests.forEach(async (test) => {
+        if (test.isInvestigated) {
+          const response = await destroy(
+            `${getProjectUrl(
```

Comment:
Same here with generating the URL just once outside the for loop.

Reply:
We are requesting on a new URL each time
Since the test Id is change.
So, this has to be sone